app/application/usecase/account/get_balance.go

```diff
@@ -2,19 +2,25 @@ package account
 
 import (
 	"context"
+	"errors"
 	customError "stoneBanking/app/domain/errors"
 	"stoneBanking/app/domain/types"
 )
 
 func (usecase *usecase) GetBalance(ctx context.Context, accountID string) (float64, error) {
 	const operation = "Usecase.Account.GetBalance"
 
-	tempAccount, err := usecase.accountRepository.GetByID(ctx, types.ExternalID(accountID))
+	tempResult, err := usecase.accountRepository.GetBalanceByAccountID(ctx, types.ExternalID(accountID))
 	if err != nil {
+		if errors.Is(err, customError.ErrorAccountIDNotFound) {
+			usecase.logger.LogError(operation, err.Error())
+			return -1, customError.ErrorAccountIDNotFound
+		}
+
 		usecase.logger.LogError(operation, err.Error())
-		return -1, customError.ErrorAccountIDNotFound
+		return -1, customError.ErrorAccountIDSearching
 	}
 
 	usecase.logger.LogInfo(operation, "balance sucessfully listed")
-	return tempAccount.Balance.ToFloat(), nil
+	return tempResult.ToFloat(), nil
```

Comment:
Outra coisa, o balance está sendo tratado como float.

No desafio tem um material explicando o problema de se usar o float:

Floating Point Math [](https://0.30000000000000004.com/)

Reply:
Na real Bruno, o balance só está sendo tratado como float para a saída, em si o Balance está sendo considerado no banco, e internamente como int.

Mas se acha que converter ele para float desde a query (considerei que não era um problema por ser uma função especializada, apenas focada em trazer o saldo, logo, não vi problema em ela já retornar no formato necessário para a exibição), posso passar essa conversão para outra camada sem problema.

Reply:
Aqui vc está na camada de usecase. Uma coisa é ser retornado como resposta de uma rota como float, outra é na sua camada de negócio. Onde vc for utilizar essa função vc vai trabalhar com float, e no link mostra a desvantagem de usá-lo

Reply:
Alteração feita